.github/workflows/nightly.yml

```diff
@@ -162,3 +162,52 @@ jobs:
           ARM_CLIENT_ID: ${{ secrets.ARM_CLIENT_ID }}
           ARM_TENANT_ID: ${{ secrets.ARM_TENANT_ID }}
           DATABRICKS_SERVERLESS_COMPUTE_ID: ${{ env.DATABRICKS_SERVERLESS_COMPUTE_ID }}
+
+  benchmark:
+    environment: tool
+    runs-on: larger
+    env:
+      DATABRICKS_SERVERLESS_COMPUTE_ID: auto
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.11'
+          cache: pip
+          cache-dependency-path: '**/pyproject.toml'
+
+      - name: Detect existing baseline
+        id: detect
+        shell: bash
+        run: |
+          shopt -s nullglob
+          files=(.benchmarks/*/0001_*.json)
+          if (( ${#files[@]} == 0 )); then
+            echo "missing=true"  >> $GITHUB_OUTPUT
+          else
+            echo "missing=false" >> $GITHUB_OUTPUT
+            echo "file=${files[0]}" >> $GITHUB_OUTPUT
+          fi
+
+      # Create baseline if missing (first run)
+      - name: Generate initial baseline
+        if: steps.detect.outputs.missing == 'true'
+        run: hatch run pytest tests/perf/test_apply_checks.py --benchmark-save=benchmark
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          ARM_CLIENT_ID: ${{ secrets.ARM_CLIENT_ID }}
+          ARM_TENANT_ID: ${{ secrets.ARM_TENANT_ID }}
+          DATABRICKS_SERVERLESS_COMPUTE_ID: ${{ env.DATABRICKS_SERVERLESS_COMPUTE_ID }}
+
+      - name: Commit baseline to repo
+        if: steps.detect.outputs.missing == 'true'
+        run: |
+          git config user.name  "github-actions[bot]"
+          git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git add .benchmarks/*/0001_*.json
+          git commit -m "Add pytest-benchmark baseline from Nightly build" || true
+          git push
```

Comment:
this will not work as we cannot push to the main directly

Reply:
Ah, okay - I think generating the first baseline locally could work, but I'm concerned about where the baseline is stored. The path is machine-dependent, so it varies depending on the system it's run on (I'm still researching how to customize this).

For example, on my machine, it's stored under: `dqx/.benchmarks/Darwin-CPython-3.10-64bit`

If someone else runs it on a different system, the baseline folder will be named differently. That means when comparing, there’s a chance the runner won’t find the baseline because the directory structure won’t match. 

I tried modifying the baseline path locally and I tried to compare again and I got this 
<img width="1677" height="247" alt="image" src="https://github.com/user-attachments/assets/57517bf2-56d5-4e15-b24c-03ea88d16fc5" />